grafana/provider.go

```diff
@@ -159,87 +255,14 @@ func Provider(version string) func() *schema.Provider {
-				"grafana_alert_notification":      ResourceAlertNotification(),
-				"grafana_builtin_role_assignment": ResourceBuiltInRoleAssignment(),
-				"grafana_contact_point":           ResourceContactPoint(),
-				"grafana_dashboard":               ResourceDashboard(),
-				"grafana_dashboard_permission":    ResourceDashboardPermission(),
-				"grafana_data_source":             ResourceDataSource(),
-				"grafana_data_source_permission":  ResourceDatasourcePermission(),
-				"grafana_folder":                  ResourceFolder(),
-				"grafana_folder_permission":       ResourceFolderPermission(),
-				"grafana_library_panel":           ResourceLibraryPanel(),
-				"grafana_message_template":        ResourceMessageTemplate(),
-				"grafana_mute_timing":             ResourceMuteTiming(),
-				"grafana_notification_policy":     ResourceNotificationPolicy(),
-				"grafana_organization":            ResourceOrganization(),
-				"grafana_playlist":                ResourcePlaylist(),
-				"grafana_report":                  ResourceReport(),
-				"grafana_role":                    ResourceRole(),
-				"grafana_rule_group":              ResourceRuleGroup(),
-				"grafana_team":                    ResourceTeam(),
-				"grafana_team_preferences":        ResourceTeamPreferences(),
-				"grafana_team_external_group":     ResourceTeamExternalGroup(),
-				"grafana_service_account_token":   ResourceServiceAccountToken(),
-				"grafana_service_account":         ResourceServiceAccount(),
-				"grafana_user":                    ResourceUser(),
-
-				// Cloud
-				"grafana_cloud_api_key":             ResourceCloudAPIKey(),
-				"grafana_cloud_plugin_installation": ResourceCloudPluginInstallation(),
-				"grafana_cloud_stack":               ResourceCloudStack(),
-
-				// Synthetic Monitoring
-				"grafana_synthetic_monitoring_check":        ResourceSyntheticMonitoringCheck(),
-				"grafana_synthetic_monitoring_probe":        ResourceSyntheticMonitoringProbe(),
-				"grafana_synthetic_monitoring_installation": ResourceSyntheticMonitoringInstallation(),
-
-				// Machine Learning
-				"grafana_machine_learning_job": ResourceMachineLearningJob(),
-
-				// OnCall
-				"grafana_oncall_integration":      ResourceOnCallIntegration(),
-				"grafana_oncall_route":            ResourceOnCallRoute(),
-				"grafana_oncall_escalation_chain": ResourceOnCallEscalationChain(),
-				"grafana_oncall_escalation":       ResourceOnCallEscalation(),
-				"grafana_oncall_on_call_shift":    ResourceOnCallOnCallShift(),
-				"grafana_oncall_schedule":         ResourceOnCallSchedule(),
-				"grafana_oncall_outgoing_webhook": ResourceOnCallOutgoingWebhook(),
-			},
+			ResourcesMap: mergeResourceMaps(
+				// Special case, this resource supports both Grafana and Cloud (depending on context)
+				map[string]*schema.Resource{
+					"grafana_api_key": ResourceAPIKey(),
+				},
+				grafanaClientResources, smClientResources, onCallClientResources, cloudClientResources),
```

Comment:
Nit: how about placing each group in separated lines? This way if in the future we need to add another group the diff will be easier to read.
```suggestion
				grafanaClientResources,
				smClientResources,
				onCallClientResources,
				cloudClientResources,
			),
```